geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/conf/AbstractArgResolver.scala

```diff
@@ -0,0 +1,94 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.conf
+
+import com.typesafe.config.ConfigParseOptions
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.annotation.tailrec
+
+/**
+ * Abstract trait for resolving sft/config parameters and handling errors around
+ * parsing.
+ */
+trait AbstractArgResolver[returnType, parseMethodArgTuple] extends LazyLogging {
+
+  // Object for holding relevant error to return
+  object ErrorData {
+    var message: String = _
+    var error: Throwable = _
+    def apply(msg: String, e: Throwable) = { message = msg; error = e }
+  }
+
+  // Type Names for arg type guesses
+  object ArgTypes extends Enumeration {
+    val  NAME, SPECSTR, CONFSTR, PATH = Value
+  }
+  import ArgTypes._
+
+  type resEither = Either[(String, Throwable, Value), returnType]
+  type argTuple = parseMethodArgTuple
```

Comment:
I use argTuple in the classes that inherit this one. For some reason the type parseMethodArgTuple isn't available there.